### PR TITLE
Implement Min Health Restriction

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
- Health is raised if below `recommendedHealth` 
- Added `maximumHealth` to schemas.

### CODE REVIEW

File: app/lib/world/world.coffee [hunk 1]
```diff
@@ -260,6 +260,11 @@ module.exports = class World
         system.start @thangs
       catch err
         console.error "Error starting system!", system, err
+    # Silently clamp health after systems run.
+    hero = _.find(@thangs, (thang) -> thang.id == 'Hero Placeholder')
```
Comment (AndrewJakubowicz) [inline app/lib/world/world.coffee]: Not sure if this is the best way to locate the hero.

File: app/lib/world/world.coffee [hunk 2]
```diff
@@ -260,6 +260,11 @@ module.exports = class World
...
+    if hero? and level.recommendedHealth?
```
Comment (AndrewJakubowicz) [inline app/lib/world/world.coffee]: If this looks good, I can implement `maximumHealth` here as well.
Not sure if it is worth refactoring out this into a World method yet. What do you think?

File: app/lib/world/world.coffee [hunk 3]
```diff
@@ -260,6 +260,14 @@ module.exports = class World
         system.start @thangs
       catch err
         console.error "Error starting system!", system, err
+    # Silently clamp health after systems run.
+    hero = _.find(@thangs, (thang) -> thang.id == 'Hero Placeholder')
+    if hero?
```
Comment (AndrewJakubowicz) [inline app/lib/world/world.coffee]: Should this be refactored out?
It's not long enough and doesn't need to be called anywhere else currently.

File: app/lib/world/world.coffee (hunk 1 above)
Comment (nwinter) [reply app/lib/world/world.coffee]: This works. You can also shorten a bit:
```
    hero = _.find @thangs, id: 'Hero Placeholder'
```

File: app/lib/world/world.coffee (hunk 3 above)
Comment (nwinter) [reply app/lib/world/world.coffee]: I would just put it into its own `world` method here called `clampHeroHealth` or something like that.

File: app/lib/world/world.coffee [hunk 4]
```diff
@@ -260,6 +260,14 @@ module.exports = class World
...
+      if level.recommendedHealth?
+        hero.maxHealth = Math.max(hero.maxHealth, level.recommendedHealth)
+      if level.maximumHealth?
+        hero.maxHealth = Math.min(hero.maxHealth, level.maximumHealth)
+      hero.health = hero.maxHealth
```
Comment (nwinter) [inline app/lib/world/world.coffee]: I just realized we only want to do this if we are in classroom mode, so when `level.type is 'course'`. In home mode, they still need to actually have that much health.

**Pull request merged**